core/src/main/scala/io/chrisdavenport/shellfish/FilesOs.scala

```diff
@@ -0,0 +1,281 @@
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package io.chrisdavenport.shellfish
+
+import cats.syntax.all.*
+import cats.effect.IO
+
+import fs2.{Stream, Chunk}
+import fs2.interop.scodec.*
+import fs2.io.file.{Files, Path, Flags}
+
+import scodec.{Codec, codecs}
+import scodec.bits.ByteVector
+
+object FilesOs {
+
+  val files = Files[IO]
+
+  // Read operations:
+
+  /**
+   * Reads the contents of the file at the path and returns it as a String.
+   * @param path
+   *   The path to read from
+   * @return
+   *   The file loaded in memory as a String
+   */
+  def read(path: Path): IO[String] = files.readUtf8(path).compile.string
+
+  /**
+   * Reads the contents of the file at the path and returns it as a ByteVector.
+   * @param path
+   *   The path to read from
+   * @return
+   *   The file loaded in memory as a ByteVector
+   */
+  def readBytes(path: Path): IO[ByteVector] =
+    files.readAll(path).compile.to(ByteVector)
+
+  /**
+   * Reads the contents of the file at the path and returns it line by line as a
+   * Vector of Strings.
+   *
+   * @param path
+   *   The path to read from
+   * @return
+   *   The file loaded in memory as a collection of lines of Strings
+   */
+  def readLines(path: Path): IO[Vector[String]] =
+    files.readUtf8Lines(path).compile.toVector
+
+  /**
+   * Reads the contents of the file and loads it as a type `A` using the
+   * provided codec.
+   * @param A
+   *   The type to read the file as
+   * @param path
+   *   The path to read from
+   * @param codec
+   *   The codec that translates the file contents into the type `A`
+   * @return
+   *   The file loaded in memory as a type `A`
+   * @throws java.lang.Exception
+   *   if the file cannot be read or the contents cannot be parsed into the type
+   *   `A`
+   */
+  def readAs[A](path: Path)(implicit codec: Codec[A]): IO[A] =
+    readBytes(path)
+      .map(bytes => codec.decodeValue(bytes.bits).require)
+      .handleErrorWith { e =>
+        new Exception(s"Failed to read file at $path", e).raiseError[IO, A]
+      }
+
+  /**
+   * The function reads the contents of the file at the path and returns it as a
+   * Stream of Bytes, useful when working with large files.
+   * @param path
+   *   The path to read from
+   * @return
+   *   A Stream of Bytes
+   */
+  def stream(path: Path): Stream[IO, Byte] = files.readAll(path)
+
+  /**
+   * The function reads the contents of the file at the `path` and returns it as
+   * a `Chunk[Bytes]`,
+   * @param path
+   *   The path to read from
+   * @return
+   *   The file loaded in memory as a Chunk of Bytes
+   */
+  def chunk(path: Path): IO[Chunk[Byte]] = files.readAll(path).compile.to(Chunk)
```

Comment:
I'd skip this method entirely. Essentially the same as `read` and folks can always convert between `ByteVector` and `Chunk[Byte]`.